wire-ios-request-strategy/Sources/Message Sending/MessageSender.swift

```diff
@@ -91,22 +95,34 @@ public class MessageSender: MessageSenderInterface {
         do {
             try await messageDependencyResolver.waitForDependenciesToResolve(for: message)
             try await attemptToSend(message: message)
+        } catch let networkError as NetworkError {
+            if case .invalidRequestError(let responseFailure, _) = networkError, responseFailure.code == 533 {
+                switch responseFailure.data?.type {
+                case .federation:
+                    responseFailure.updateExpirationReason(for: message, with: .federationRemoteError)
```

Comment:
what does this do? isn't it message a CoreData entity?

Reply:
Message is often a CoreData entity but not always. I just copied this code from the previous implementation I guess it record why a message couldn't be sent when some federated backend couldn't be reached.

Reply:
if expirationReasonCode is a CoreData prop, we need to make sure we're on the right context as we're on an async function here, but I guess it can be updateExpirationReason implementation details

Reply:
Yep you're right, fixed it.